File: base/constants/status.verification.go
package constants

import "reflect"

type VerificationStatus struct {
	Pending             string
	PendingVerification string
	Verified            string
	Rejected            string
	Escalated           string
	Expired             string
	Cancelled           string
	Missing             string
}

var VerificationStatuses = VerificationStatus{
	Pending:             "PEND",
	PendingVerification: "PEVF",
	Verified:            "VERF",
	Rejected:            "RJCT",
	Escalated:           "ESCL",
	Expired:             "EXPD",
	Cancelled:           "CNCL",
	Missing:             "MISS",
}

func GetAllVerificationStatuses() []string {
	v := reflect.ValueOf(VerificationStatuses)
	var statuses []string

	for i := 0; i < v.NumField(); i++ {
		statuses = append(statuses, v.Field(i).String())
	}

	return statuses
}

File: ui/src/interfaces/status.verification.ts
//||------------------------------------------------------------------------------------------------||
//|| Types
//||------------------------------------------------------------------------------------------------||

export type VerificationStatusKeys =
      | "None"
      | "Pending"
      | "Verified"
      | "Failed";

export type VerificationStatusMap = Record<VerificationStatusKeys, string>;

//||------------------------------------------------------------------------------------------------||
//|| Constants
//||------------------------------------------------------------------------------------------------||

export const VerificationStatuses: VerificationStatusMap = {
      None:     "NONE",
      Pending:  "PEND",
      Verified: "VERF",
      Failed:   "FAIL",
};

//||------------------------------------------------------------------------------------------------||
//|| Utils
//||------------------------------------------------------------------------------------------------||

export function getAllVerificationStatuses(): string[] {
      return Object.values(VerificationStatuses);
}

export function getVerificationStatusMap(): Record<string, string> {
      return { ...VerificationStatuses };
}
